cdc/owner/async_sink.go

```diff
@@ -34,94 +33,109 @@ const (
 	defaultErrChSize = 1024
 )
 
-// AsyncSink is an async sink design for owner
+// asyncSink is an async sink design for owner
 // The EmitCheckpointTs and EmitDDLEvent is asynchronous function for now
 // Other functions are still synchronization
-type AsyncSink interface {
-	// EmitCheckpointTs emits the checkpoint Ts to downstream data source
+type asyncSink interface {
+	// run the asyncSink
+	run(ctx cdcContext.Context, id model.ChangeFeedID, info *model.ChangeFeedInfo) error
+	// emitCheckpointTs emits the checkpoint Ts to downstream data source
 	// this function will return after recording the checkpointTs specified in memory immediately
 	// and the recorded checkpointTs will be sent and updated to downstream data source every second
-	EmitCheckpointTs(ctx cdcContext.Context, ts uint64)
-	// EmitDDLEvent emits DDL event asynchronously and return true if the DDL is executed
+	emitCheckpointTs(ctx cdcContext.Context, ts uint64)
+	// emitDDLEvent emits DDL event asynchronously and return true if the DDL is executed
 	// the DDL event will be sent to another goroutine and execute to downstream
 	// the caller of this function can call again and again until a true returned
-	EmitDDLEvent(ctx cdcContext.Context, ddl *model.DDLEvent) (bool, error)
-	SinkSyncpoint(ctx cdcContext.Context, checkpointTs uint64) error
-	Close(ctx context.Context) error
+	emitDDLEvent(ctx cdcContext.Context, ddl *model.DDLEvent) (bool, error)
+	emitSyncPoint(ctx cdcContext.Context, checkpointTs uint64) error
+	close(ctx context.Context) error
 }
 
 type asyncSinkImpl struct {
-	sink           sink.Sink
-	syncpointStore sink.SyncpointStore
+	lastSyncPoint  model.Ts
+	syncPointStore sink.SyncpointStore
 
-	checkpointTs model.Ts
-
-	lastSyncPoint model.Ts
-
-	ddlCh         chan *model.DDLEvent
+	checkpointTs  model.Ts
 	ddlFinishedTs model.Ts
 	ddlSentTs     model.Ts
 
+	ddlCh chan *model.DDLEvent
+	errCh chan error
+
+	sink sink.Sink
+	// `sinkInitHandler` can be helpful in unit testing.
+	sinkInitHandler asyncSinkInitHandler
+
 	cancel context.CancelFunc
-	wg     sync.WaitGroup
```

Comment:
We should leave it unchanged, it encapsulates asynchronicity very well.

Reply:
cc: @3AceShowHand PTAL~

Reply:
I agreed with rustin.

Reply:
I would prefer to rename it `DDLSink`.